diff -Naru a/xbmc/utils/MathUtils.h b/xbmc/utils/MathUtils.h
--- a/xbmc/utils/MathUtils.h	2013-12-12 22:47:49.000000000 +0100
+++ b/xbmc/utils/MathUtils.h	2014-05-02 14:31:45.825800514 +0200
@@ -27,7 +27,7 @@
 #ifdef __SSE2__
 #include <emmintrin.h>
 #endif
-
+#define DISABLE_MATHUTILS_ASM_ROUND_INT        1
 // use real compiler defines in here as we want to
 // avoid including system.h or other magic includes.
 // use 'gcc -dM -E - < /dev/null' or similar to find them.
